docs/creating-test-environment.md

```diff
@@ -0,0 +1,194 @@
+```
+
+### Use External Clusters
+
+Make sure to have two `KUBECONFIG` compliant client configurations at hand, one for kcp, one for skr
+
+## Setting up your registry
+
+### Make sure the registries are reachable via localhost
+
+Add the following to your `etc/hosts` entry.
+
+```/etc/hosts
+##
+# Host Database
+#
+# localhost is used to configure the loopback interface
+# when the system is booting.  Do not change this entry.
+##
+127.0.0.1       localhost
+255.255.255.255 broadcasthost
+::1             localhost
+
+# Added by Docker Desktop
+# To allow the same kube context to work on the host and the container:
+127.0.0.1 kubernetes.docker.internal
+
+# Added for Operator Registries
+127.0.0.1 op-kcp-registry.localhost
+127.0.0.1 op-skr-registry.localhost
+```
+
+### Using an external Registry
+
+TODO
+
+## Make sure you are in the Control Plane
+
+```
+kubectl config use k3d-op-kcp
+```
+
+## Install CRDs of Operator Stack
+
+_Note for Remote Clusters: Make sure you run the commands with `KUBECONFIG` set to the KCP Cluster_
+
+### Install Manifest Operator CRDs
+
+1. Checkout https://github.com/kyma-project/manifest-operator
+2. Run the Installation Command
+
+```
+make install
+```
+
+### Install Kyma Operator CRDs
+
+1. Checkout https://github.com/kyma-project/kyma-operator and `cd operator`
+2. Run the Installation Command
+
+```
+make install
+```
+
+Ensure the CRDs are installed with `k get crds | grep kyma-project.io`:
+
+```
+manifests.component.kyma-project.io        2022-08-18T16:27:21Z
+kymas.operator.kyma-project.io             2022-08-18T16:29:28Z
+moduletemplates.operator.kyma-project.io   2022-08-18T16:29:28Z
+```
+
+## Build your module
+
+In `https://github.com/kyma-project/kyma-operator`, `cd samples/template-operator`
+
+After this find the Port of your KCP OCI Registry and write it to `MODULE_REGISTRY_PORT`:
+
+### Using a Local Module/Image Registry
+
+```
+export MODULE_REGISTRY_PORT=$(docker port op-kcp-registry.localhost 5000/tcp | cut -d ":" -f2)
+export IMG_REGISTRY_PORT=$(docker port op-skr-registry.localhost 5000/tcp | cut -d ":" -f2)
+```
+
+### Using a Remote Module/Image Registry
+
+```
+export MODULE_REGISTRY=your-registry-goes-here.com
+export IMG_REGISTRY=your-registry-goes-here.com
+```
+
+### Generating and Pushing the Operator Image and Charts
+
+Next generate and push the module image of the operator
+
+```
+make module-operator-chart module-image
+```
+
+## Use your module and trigger a Kyma Installation
+
+_Note for Remote Clusters: Make sure you run the commands with `KUBECONFIG` set to the KCP Cluster_
+
+First make sure that the `kyma-system` namespace is created:
+
+```
+kubectl create ns kyma-system
+```
+
+After this, build and push the module the module to the Registry with
+
+```
+make module-build module-template-push
+```
+
+Before we start reconciling, lets create a secret to access the SKR:
+
+In https://github.com/kyma-project/kyma-operator run
```

Comment:
This must be executed from within the `./operator` subdirectory, or the path in the command must contain the `operator/` prefix